grid2op/tests/test_ma.py

```diff
@@ -597,6 +608,311 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
                 raise AssertionError("some interco are deactivated in the mask pos topo vect")
             
+    
+    def check_action_spaces(self, ma_env):
+        # This function checks if the action space is correctly
+        # created from _subgrids_cls['action']
+        # TODO more tests 
+        for agent in ma_env.agents:
+            # We test if they have the same topological dimension
+            assert ma_env.action_spaces[agent].dim_topo\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].dim_topo
+                    
+            # It checks if the number of True values in the mask is
+            # equal to the length of actions' _set_topo_vect
+            assert len(ma_env.action_spaces[agent]({})._set_topo_vect)\
+                ==\
+                    np.sum(ma_env._subgrids_cls['action'][agent].mask_orig_pos_topo_vect)
+                    
+            # Checks if they both have the same load_pos_topo_vect
+            assert (ma_env.action_spaces[agent].load_pos_topo_vect\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].load_pos_topo_vect).all()
+            
+            # Checks if they both have the same gen_pos_topo_vect
+            assert (ma_env.action_spaces[agent].gen_pos_topo_vect\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].gen_pos_topo_vect).all()
+            
+            # Checks if they both have the same line_or_pos_topo_vect
+            assert (ma_env.action_spaces[agent].line_or_pos_topo_vect\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].line_or_pos_topo_vect).all() 
+                    
+            # Checks if they both have the same line_ex_pos_topo_vect
+            assert (ma_env.action_spaces[agent].line_ex_pos_topo_vect\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].line_ex_pos_topo_vect).all() 
+                    
+            # Checks if they both have the same storage_pos_topo_vect
+            assert (ma_env.action_spaces[agent].storage_pos_topo_vect\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].storage_pos_topo_vect).all() 
+            
+            # Checks if they both have the same interco_pos_topo_vect
+            assert (ma_env.action_spaces[agent].interco_pos_topo_vect\
+                ==\
+                    ma_env._subgrids_cls['action'][agent].interco_pos_topo_vect).all() 
+        
+    def are_same_actions(self, a1, a2, type_action, 
+                         add_msg = ""):
+        # Checks both actions have the same action type
+        # meaning if the same 
+        assert (getattr(a1, type_action) == getattr(a2, type_action)).all(), add_msg
+        # Chacks if the same action type is applied 
+        assert getattr(a1, f'_modif_{type_action}') == getattr(a2, f'_modif_{type_action}'), add_msg
```

Comment:
you copy pasted something twice here ! have a variable that represents `f'_modif_{type_action}'`

Reply:
done